modules/openstack/service.go

```diff
@@ -39,24 +39,44 @@ func (o *OpenStack) CreateService(
 	log logr.Logger,
 	s Service,
 ) (string, error) {
-
-	createOpts := services.CreateOpts{
-		Type:    s.Type,
-		Enabled: &s.Enabled,
-		Extra: map[string]interface{}{
-			"name":        s.Name,
-			"description": s.Description,
-		},
+	var serviceID string
+
+	service, err := o.GetService(
+		log,
+		s.Type,
+		s.Name,
+	)
+	if err != nil && !strings.Contains(err.Error(), ServiceNotFound) {
+		return serviceID, err
 	}
 
-	service, err := services.Create(o.GetOSClient(), createOpts).Extract()
-	if err != nil {
-		return "", err
+	// if there is already a service, use it
+	if service != nil {
+		serviceID = service.ID
+	} else {
+		createOpts := services.CreateOpts{
+			Type:    s.Type,
+			Enabled: &s.Enabled,
+			Extra: map[string]interface{}{
+				"name":        s.Name,
+				"description": s.Description,
+			},
+		}
+
+		service, err := services.Create(o.GetOSClient(), createOpts).Extract()
+		if err != nil {
+			return serviceID, err
+		}
+		log.Info(fmt.Sprintf("Service Created - Servicename %s, ID %s", s.Name, service.ID))
+		serviceID = service.ID
 	}
 
-	return service.ID, nil
+	return serviceID, nil
 }
 
+// ServiceNotFound - service not found error message"
+const ServiceNotFound = "service not found in keystone"
+
```

Comment:
Should we maybe put this at the top somewhere for organizational/readability purposes?

Reply:
done